nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/QueryElasticsearchHttp.java

```diff
@@ -175,16 +197,30 @@
             .allowableValues(TARGET_FLOW_FILE_CONTENT, TARGET_FLOW_FILE_ATTRIBUTES)
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR).build();
 
-    private static final Set<Relationship> relationships;
+    public static final PropertyDescriptor ROUTING_QUERY_INFO_STRATEGY = new PropertyDescriptor.Builder()
+            .name("routing-query-info-strategy")
+            .displayName("Routing Strategy for Query Info")
+            .description("Specifies when to generate and route Query Info after a successful query")
+            .expressionLanguageSupported(false)
+            .allowableValues(ALWAYS, NEVER, NO_HITS)
+            .defaultValue(NEVER.getValue())
+            .required(false)
+            .build();
+
+    public static final PropertyDescriptor INCLUDE_QUERY_IN_ATTRS = new PropertyDescriptor.Builder()
```

Comment:
Should we really need to have a property for this? Writing the `es.query.url` as an attribute by default sounds safe to me. Thoughts?

Reply:
@mattyb149  thoughts?

Reply:
Having the attribute by default sounds fine to me, usually when we have a property to choose it's because the value in the attribute could be fairly large (such as a compound SQL query used in ExecuteSQL, e.g.).